test/components/bitvector.cpp

```diff
@@ -0,0 +1,211 @@
+// SPDX-FileCopyrightText: 2017 - 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+/*@GKO_PREPROCESSOR_FILENAME_HELPER@*/
+
+#include "core/components/bitvector.hpp"
+
+#include <cassert>
+#include <memory>
+#include <random>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/base/array.hpp>
+#include <ginkgo/core/base/executor.hpp>
+
+#include "common/unified/base/kernel_launch.hpp"
+#include "common/unified/components/bitvector.hpp"
+#include "core/base/index_range.hpp"
+#include "core/test/utils.hpp"
+#include "reference/components/bitvector.hpp"
+#include "test/utils/common_fixture.hpp"
+
+
+// workaround for cudafe 11.0 bug
+using gko::irange;
+
+
+template <typename T>
+class Bitvector : public CommonTestFixture {
+protected:
+    using index_type = T;
+    using bitvector = gko::bitvector<index_type>;
+    using device_bitvector = gko::device_bitvector<index_type>;
+    using storage_type = typename bitvector::storage_type;
+    constexpr static auto block_size = bitvector::block_size;
+
+    Bitvector()
+        : rng{67193}, sizes{0,    1,    2,    16,    31,    32,  33,
+                            40,   63,   64,   65,    127,   128, 129,
+                            1000, 1024, 2000, 10000, 100000}
+    {}
+
+    gko::array<index_type> create_random_values(index_type num_values,
+                                                index_type size)
+    {
+        assert(num_values <= size);
+        std::vector<index_type> values(size);
+        std::iota(values.begin(), values.end(), index_type{});
+        std::shuffle(values.begin(), values.end(), rng);
+        values.resize(num_values);
+        std::sort(values.begin(), values.end());
+        return gko::array<index_type>{this->ref, values.begin(), values.end()};
+    }
+
+    void assert_bitvector_equal(const bitvector& bv, const bitvector& dbv)
+    {
+        ASSERT_EQ(bv.get_size(), dbv.get_size());
+        const auto num_blocks =
+            static_cast<gko::size_type>(bv.get_num_blocks());
+        const auto bits =
```

Comment:
we have also dealed the const_array_view in `GKO_ASSERT_ARRAY_EQ`, so you do not need to cast them back to non-const